enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -310,15 +310,20 @@ func (s *ExecutionServer) updateExecution(ctx context.Context, executionID strin
 		}
 	}
 
-	var dbErr error
-	var existing tables.Execution
-	// TODO(zoey): This select seems unnecessary, should revisit it
-	if err := s.env.GetDBHandle().GORM(ctx, "execution_server_get_execution_for_update").Where(
-		"execution_id = ?", executionID).First(&existing).Error; err != nil {
-		dbErr = err
-	} else {
-		dbErr = s.env.GetDBHandle().GORM(ctx, "execution_server_update_execution").Model(&existing).Where(
-			"execution_id = ? AND stage != ?", executionID, repb.ExecutionStage_COMPLETED).Updates(execution).Error
+	result := s.env.GetDBHandle().GORM(ctx, "execution_server_update_execution").Where(
+		"execution_id = ? AND stage != ?", executionID, repb.ExecutionStage_COMPLETED).Updates(execution)
+	dbErr := result.Error
+	if dbErr == nil && result.RowsAffected == 0 {
+		// We want to return an error if the execution simply doesn't exist, but
+		// we want to ignore any attempts to update a cancelled execution.
+		var count int64
+		if err := s.env.GetDBHandle().GORM(
+			ctx,
```

Comment:
We could use this opportunity to replace this code with a simple "SELECT count(*) ..." query. What do you think?

Reply:
Sure, sounds good to me! No real reason to use the ORM if we aren't leveraging the multi-SQL-language stuff or the automatic columns-to-data-fields mapping, after all ^-^